activities/activity_laginhas.py

```diff
@@ -0,0 +1,45 @@
+#!/usr/bin/env python3
+
+"""
+Comparing two different resample interpolation methods: nearest neighbor and bilinear.
```

Comment:
Another issue is, what are you trying to show with this on Tangible Landscape? It needs to have some connection to topography, so that when you change topography you can see change in your analysis. Here, the largest differences would be (I think) connected to high slope areas. But it won't be that much visible from the difference itself. Perhaps running moving window analysis (r.neighbors) with range on the differences would highlight the areas with largest errors?

Reply:
Yes - originally, I set the resolution to 100m at the beginning of the script. So when I used the resampled interpolation it would resample at the specified higher resolution. This was before I realized that TL requires a res of 4m. Although it isn't a drastic change (resample from 1m to 4m rather than 1m to 100m), the script still aims to do what I intended, which was show the difference between a map resampled at a coarser resolution using the nearest neighbor methods vs. bilinear method. 

I have attached a photo of my output. The purpose of this script was to visualize the the spatial distribution of error that could be introduced by choosing different interpolation methods when resampling to a coarser resolution.  Does this make sense?
<img width="599" alt="Screen Shot 2021-04-19 at 10 46 59 AM" src="https://user-images.githubusercontent.com/67606742/115255766-95c74780-a0fc-11eb-8e24-8718a201c6fa.png">

Reply:
> Yes - originally, I set the resolution to 100m at the beginning of the script. So when I used the resampled interpolation it would resample at the specified higher resolution. This was before I realized that TL requires a res of 4m. Although it isn't a drastic change (resample from 1m to 4m rather than 1m to 100m), the script still aims to do what I intended, which was show the difference between a map resampled at a coarser resolution using the nearest neighbor methods vs. bilinear method.

I understand what you are trying to do, but as it is now, the result depends on current region setting which is arbitrary. When running TL applications, I always pass the environment with the region setting matching the scanned elevation raster (the resolution and extent are slightly different for each scan). If you don't do that, it runs with whatever region is currently set, but it won't match. So yes, your application still works, but I am trying to have more under control which resolution you are actually resampling to. Building on top of what I suggested earlier, I would take the resolution of the raster and divide it by e.g. 2 and then use that for resampling.
```
raster_region = gs.region(env=env)
res = (raster_region['nsres'] + raster_region['ewres']) / 2
env2 = env.copy()
env2['GRASS_REGION'] = gs.region_env(res=res / 2)
``` 



> 
> I have attached a photo of my output. The purpose of this script was to visualize the the spatial distribution of error that could be introduced by choosing different interpolation methods when resampling to a coarser resolution. Does this make sense?
> <img alt="Screen Shot 2021-04-19 at 10 46 59 AM" width="599" src="https://user-images.githubusercontent.com/67606742/115255766-95c74780-a0fc-11eb-8e24-8718a201c6fa.png">

Yes, and this works too, I just felt the effect of topography is not very visible here, but I will leave that up to you.

Reply:
This is definitely a more flexible function and what I ultimately wanted to do - thank you. 

I want a coarser resolution, so should I multiply instead of dividing?

One more clarifying question, when you mention that the effect of topography is not visible here, do you mean for this particular area? If yes, I wonder if I should take the absolute values of the elev_diff and then display that instead?  

Thanks again!

Reply:
Well, using the wrong function can certainly be a problem. I updated the function to resample.stats. Hopefully the line of code I entered to increase the resolution prior to resampling resolves the other issues you brought up. Thanks again!